tools/dependencies/check-LICENSE.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# List all modules(jars) that belong to the SkyWalking itself, these will be ignored
+# when checking the dependency licenses
+./mvnw -Pbackend -Dexec.executable='echo' -Dexec.args='${project.artifactId}-${project.version}.jar' exec:exec -q > self-modules.txt
+
+ls dist/apache-skywalking-apm-bin/oap-libs > all-dependencies.txt
+
+# Exclude all self modules(jars) to generate all third-party dependencies
+grep -vf self-modules.txt all-dependencies.txt > third-party-dependencies.txt
+
+# Compare the third-party dependencies with known dependencies, expect that
+# all third-party dependencies are KNOWN and the exit code of the command is 0,
+# otherwise we should add its license to LICENSE file and add the dependency to known-dependencies.txt
+diff -U0 tools/dependencies/known-dependencies.txt third-party-dependencies.txt
```

Comment:
You need to consider order. Maybe different in different OS.

Reply:
> You need to consider order. Maybe different in different OS.

I sort the dependencies when generating the files